goldmane/pkg/client/client.go

```diff
@@ -52,8 +68,52 @@ func (c *FlowClient) Run(ctx context.Context, grpcClient grpc.ClientConnInterfac
 	// Start the cache cleanup task.
 	go c.cache.Run(FlowCacheCleanup)
 
+	startUp := make(chan struct{})
+	go func() {
+		rc, err := c.connect(ctx)
+		// Close this regardless of the error since we don't want the other side of the channel to hang forever.
+		close(startUp)
+		if err != nil {
+			logrus.WithError(err).Warn("Unable to connect to flow server.")
+			return
+		}
+
+		// Send new Flows as they are received.
+		for flog := range c.inChan {
+			// Add the flow to our cache. It will automatically be expired in the background.
+			// We don't need to pass in a value for scope, since the client is intrinsically scoped
+			// to a particular node.
+			c.cache.Add(flog, "")
+
+			// Send the flow.
+			if err := rc.Send(&proto.FlowUpdate{Flow: flog}); err != nil {
+				logrus.WithError(err).Warn("Failed to send flow")
+				break
+			}
+
+			// Receive a receipt.
+			if _, err := rc.Recv(); err != nil {
+				logrus.WithError(err).Warn("Failed to receive receipt")
+				break
+			}
+		}
+
+		if err := rc.CloseSend(); err != nil {
+			logrus.WithError(err).Warn("Failed to close connection")
+		}
+
+		rc, err = c.connect(ctx)
```

Comment:
Why do we need another call to connect here?

Reply:
This is the reconnect logic, if that loop before this exits it means there was an error that requires a reconnect. I think when you wrote this comment I may not have had the outer loop.